hashmap.h

```diff
@@ -144,6 +144,18 @@ static int hashmap_iterate(const struct hashmap_s *const hashmap,
                            int (*f)(void *const context, void *const value),
                            void *const context) HASHMAP_UNUSED;
 
+/// @brief Iterate over all the heys and elements in a hashmap.
+/// @param hashmap The hashmap to insert into.
+/// @param f The function pointer to call on each element.
+/// @param context The context to pass as the first argument to f.
+/// @return If the entire hashmap was iterated then 0 is returned.
+/// Otherwise if the callback function f returned positive then non-zsero
```

Comment:
`non-zsero` -> `non-zero`

several issues:
- the comment overall is not very understandable
- your comment doesn't make clear that iterating stops when the callback returns a negative number

in general i would do an enum:
```
enum iterate_hint {
  ITER_CONTINUE,
  ITER_STOP,
  ITER_DELETE_CURRENT
};
```

Reply:
Enumerations with signed values are compiler dependent for their implementations.  If the underlying data type is unsigned, it will cost more than it solves.